File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.cs
using System;

public class Solution {
    public int[] solution(string s) {
        int[] answer = new int[2];
        int zeroCount=0,conversionCount=0;
        
        while(s.Length!=1)
        {
            int oneCount=0;
            
            foreach(var c in s)
            {
                if(c=='0')
                {
                    zeroCount++;
                }
                else
                {
                    oneCount++;
                }
            }
            s=Convert.ToString(oneCount,2);
            conversionCount++;
        }
        
        answer[0]=conversionCount;
        answer[1]=zeroCount;
        
        return answer;
    }
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];

        while(true){
            StringBuilder sb = new StringBuilder();
            char[] ch = s.toCharArray();
            for(int i = 0; i < ch.length; i++){
                if(ch[i] != '0'){
                    sb.append(ch[i]);
                    continue;
                }
                answer[1]++;
            }
            int len = sb.toString().length();
            String str = Integer.toString(len,2);
            
            answer[0]++;
            if(str.equals("1")){
                break;
            }
            s = str;
        }
        return answer;
    }
}